core/lib/zksync_core/src/utils/mod.rs

```diff
@@ -87,6 +87,33 @@ pub(crate) async fn projected_first_l1_batch(
     Ok(snapshot_recovery.map_or(L1BatchNumber(0), |recovery| recovery.l1_batch_number + 1))
 }
 
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct BlockStartInfo {
```

Comment:
This structure seems to be only used in the `api_server` module. Do you anticipate it being useful elsewhere? If not, maybe it'll be better to move it there?
`utils` module feels like a somewhat awkward place to put it.